### PR TITLE
fix: sorting function for date fields

### PR DESCRIPTION
**Pull request by hui00**
This Pull Request introduces sorting with HTML data-sort (timestamp-integers) and remove moment.js

1. add sort for created_at and updated_at columns
2. add sort for date_start, date_end, and to_end columns
3. remove moment-related code and dependencies

SVA-1122

### CODE REVIEW

**Comment by donni106** [comment]
You forgot POIs:

```
<td><%= toLocalDateTime(poi.updated_at) %></td>
<td><%= toLocalDateTime(poi.created_at) %></td>
```

I will adjust that.

**Comment by donni106** [comment]
And Tour:

```
<td><%= toLocalDateTime(tour.updated_at) %></td>
<td><%= toLocalDateTime(tour.created_at) %></td>
```

I will adjust as well.

File: app/helpers/application_helper.rb [hunk 1]
```diff
@@ -27,6 +27,10 @@ def toLocalDateTime(date_time)
     date_time.to_datetime.in_time_zone("Berlin").strftime("%d.%m.%Y %H:%M Uhr")
   end
 
+  def to_unix_timestamp(date)
+    date.present? ? DateTime.parse(date.to_s).to_i : 0
+  end
```
[the remark this answers is not in the document]
Comment (donni106) [reply app/helpers/application_helper.rb:32]: we have enough "space" here and do not need a short line when creating this method

File: app/javascript/packs/application.js [hunk 2]
```diff
@@ -87,16 +83,7 @@ $(function () {
...
-    language: {
-      url: '//cdn.datatables.net/plug-ins/9dcbecd42ad/i18n/German.json'
-    },
     order: [[0, 'desc']],
-    columnDefs: [
-      {
-        bSortable: false,
-        targets: 'nosort'
-      }
-    ]
```
Comment (donni106) [inline app/javascript/packs/application.js:99]: this should have not been removed! this is not related to date/time/moment, but to be able to mark some columns as not sortable. with your change this results in

image

where the button column would be sortable now.
i will revert that.

File: package.json [hunk 3]
```diff
@@ -15,7 +15,6 @@
     "@rails/webpacker": "^5.2.2",
     "axios": "^0.26.1",
     "bootstrap": "^4.6.0",
-    "datatable-sorting-datetime-moment": "^1.0.1",
```
Comment (donni106) [inline package.json:18]: you have removed the `require('moment');` next to `require('datatable-sorting-datetime-moment');` in application.js, but only removed the second one here. how come? `moment` can be removed entirely here as well, or am i wrong?

Comment (donni106) [reply package.json:18]: 52173b4

**Review by donni106 — APPROVED** [review]

File: app/javascript/packs/application.js (hunk 2 above)
Comment (hui00) [reply app/javascript/packs/application.js]: ah... thank you for the explanation

File: package.json (hunk 3 above)
Comment (hui00) [reply package.json:18]: you are right. i initially wanted to remove it but wasn't sure, so I left it

**Pull request merged**